### ISSUE TITLE
Can't join a room with room membership access over federation

### ISSUE THREAD

**Issue by GearKite** [girlbossceo/conduwuit#501]
Trying to join a room with access set to space members over federation results in:
`MatrixError: [403] Answer from example.com: [403 / M_FORBIDDEN] M_FORBIDDEN: Event is not authorized.`

Steps to reproduce:
1. Create a space
2. Create a room under that space with `Visible to space members`
3. Invite a user on another homeserver to the space
4. Attempt to join the room from the other homeserver

Doing this with 2 accounts on a single homeserver seems to work fine.

**Comment by Kimiblock** [comment]
Yeah this happens to me. If one add another space hosted on Synapse to the access list, others can join.

**Comment by yellows111** [comment]
> Yeah this happens to me. If one add another space hosted on Synapse to the access list, others can join.

What do you mean by this? This is *very* vague.

What does "add another space hosted on Synapse" imply? What's a "space" in this context? A room?

Does that mean join another room from the remote server running Synapse to a user on the Conduwuit server (local)?

**Comment by Kimiblock** [comment]
No, I mean that a remote server (only tried Synapse) cannot join a room which is hosted on conduwuit and is restricted to a space on conduwuit.

Upon adding a space hosted on Synapse to the allow list, other servers’ users can join.

This is very weird and discovered when I host some bridges for the Arch community.


--
Sincerely, 
Kimiblock

**Comment by GearKite** [comment]
> Is this still an issue?

Yup, same error on 16014e1.

**Comment by girlbossceo** [comment]
It's been a while and we've done a lot of membership refactoring which may have fixed this, is this still issue on `main` branch?

I haven't been able to reproduce it myself.

**Comment by GearKite** [comment]
Seems to work now. 🎉
